src/data_structures/fmindex.rs

```diff
@@ -553,11 +621,42 @@ mod tests {
 
         let sai = fm.backward_search(pattern.iter());
 
-        let positions = sai.occ(&sa);
+        let positions = match sai {
+            BackwardSearchResult::Complete(saint) => saint.occ(&sa),
+            BackwardSearchResult::Partial(saint, _l) => saint.occ(&sa),
+            BackwardSearchResult::Absent => Vec::<usize>::new(),
+        };
 
         assert_eq!(positions, [0]);
     }
 
+    #[test]
+    fn test_fmindex_backward_search_partial_match() {
+        let text = b"GATTACA$";
+        let pattern = b"GTACA";
+        let alphabet = dna::n_alphabet();
+        let sa = suffix_array(text);
+        let bwt = bwt(text, &sa);
+        let less = less(&bwt, &alphabet);
+        let occ = Occ::new(&bwt, 3, &alphabet);
+        let fm = FMIndex::new(&bwt, &less, &occ);
+
+        let sai = fm.backward_search(pattern.iter());
+
+        let mut partial_match_len = 0;
+        let positions = match sai {
+            BackwardSearchResult::Complete(saint) => saint.occ(&sa),
+            BackwardSearchResult::Partial(saint, l) => {
+                partial_match_len = l;
+                saint.occ(&sa)
+            }
+            BackwardSearchResult::Absent => Vec::<usize>::new(),
+        };
```

Comment:
Since we're expecting a partial match here, I think having only the corresponding Partial match arm and a panic/unreachable/... for the other two should make this (and also the other) testcase(s) clearer